src/coffee/sync/utils/product.coffee

```diff
@@ -372,7 +378,8 @@ class ProductUtils extends BaseUtils
             # no idea what this is but lets just use it for updating
             updated[k] = diffed_value
 
-      if old_obj[key]
+      # extend with an old object only if value is an object or an array
```

Comment:
Hello @hisabimbola - I added info here 7fde784 .. We did not have a `key` attribute in the base object before so there were all attributes with a type object and this code was extending new objects with the old ones.

But now we have also attribute 'key' with a type string which can't be extended so I put there this condition which will just use the new version.